### ISSUE TITLE
pronto-rubocop crashes with undefined method `rewrite' for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by d4rky-pl** [prontolabs/pronto-rubocop#83]
I'm investigating an issue after upgrading to the latest rubocop and pronto-rubocop in our project. It seems that it's sometimes crashing with

```
/path/vendor/bundle/ruby/3.0.0/gems/pronto-rubocop-0.11.4/lib/pronto/rubocop/offense_line.rb:43:in `corrected_lines': undefined method `rewrite' for nil:NilClass (NoMethodError)
```

Further investigation suggests that `RuboCop::Cop::Base::InvestigationReport` instance has `corrector` set to `nil`. This causes `corrected_lines` to fail. The cop that causes this is `RuboCop::Cop::Metrics::ClassLength` but I have no idea how to debug it further.

**Comment by d4rky-pl** [comment]
I've managed to narrow it down a bit, it seems to be working fine in rubocop 1.40.0 but starts failing in >= 1.41.0. [There seems to have been some changes around the correctors in 1.41.0](https://github.com/rubocop/rubocop/compare/v1.40.0...v1.41.0) which may cause the incompatibility.

**Comment by ashkulz** [comment]
@d4rky-pl is it possible to create a test case? If I see a [recent run](https://github.com/prontolabs/pronto-rubocop/actions/runs/3927747947/jobs/6714662341), it used rubocop 1.43.0. I'd want to enhance the tests to cover the scenario you're hitting before we can actually fix it.

**Comment by jesseclark** [comment]
I just hit this one too. It fails consistently on one PR but works fine on others.

- pronto-rubocop-0.11.4
- rubocop (1.43.0)
- rubocop-ast (1.24.1)
- rubocop-performance (1.15.2)
- rubocop-rails (2.17.4)
- rubocop-rspec (2.17.0)
- ruby 3.1.0

**Comment by ashkulz** [comment]
@jesseclark which cops are enabled? Also, is it possible to make to look at the diff and figure out which cop is triggering this?

I would love to have a small testcase which is reproducible :sweat_smile:

**Comment by jesseclark** [comment]
@ashkulz The only cop violation I see in the files that changed is `Rails/LexicallyScopedActionFilter`. 

Getting the full list of enabled/disabled cops might be a bit more involved but I could do so if that isn't enough to go on.

**Comment by jesseclark** [comment]
Just for a little extra verification, I modified one of the lines that cause the violation and opened a new PR and saw the `NoMethodError`. I then disabled `Rails/LexicallyScopedActionFilter` and pushed that. Pronto runs fine.

The code that is generating the violation is in a Rails Controller where we have a `before_action` call scope to `only: :show` but the `show` action isn't defined in the controller. If that helps you to setup a test case.

**Comment by ashkulz** [comment]
@d4rky-pl / @jesseclark, can you check if it works for you with the linked PR?

**Comment by d4rky-pl** [comment]
@ashkulz apologies, I was not able to get back to this topic until today. I can confirm the linked PR fixes the problem with rubocop > 1.40 both locally and our CI.

**Comment by ashkulz** [comment]
This has been [pushed to RubyGems](https://rubygems.org/gems/pronto-rubocop/versions/0.11.5) :tada:

**Comment by jesseclark** [comment]
Thanks @ashkulz!
